src/test/scala/is/hail/utils/RichIndexedRowMatrixSuite.scala

```diff
@@ -29,23 +31,26 @@ class RichIndexedRowMatrixSuite extends SparkSuite {
       breezeConverter.invoke(sparkMatrix).asInstanceOf[Matrix[Double]]
     }
 
-    // Tests when n % colsPerBlock != 0
-    val blockMat = idxRowMat.toBlockMatrixDense(2, 2)
-    assert(blockMat.numRows() === m)
-    assert(blockMat.numCols() === n)
-    assert(convertDistributedMatrixToBreeze(blockMat) === convertDistributedMatrixToBreeze(idxRowMat))
+    val blockMat = idxRowMat.toHailBlockMatrix(2)
+    assert(blockMat.rows === rows)
+    assert(blockMat.cols === cols)
+    assert(blockMat.toLocalMatrix() === convertDistributedMatrixToBreeze(idxRowMat))
 
-    // Tests when m % rowsPerBlock != 0
-    val blockMat2 = idxRowMat.toBlockMatrixDense(3, 1)
-    assert(blockMat2.numRows() === m)
-    assert(blockMat2.numCols() === n)
-    assert(convertDistributedMatrixToBreeze(blockMat2) === convertDistributedMatrixToBreeze(idxRowMat))
+    val blockMat2 = idxRowMat.toHailBlockMatrix(4)
+    assert(blockMat2.rows === rows)
+    assert(blockMat2.cols === cols)
+    assert(blockMat2.toLocalMatrix() === convertDistributedMatrixToBreeze(idxRowMat))
+
+    val blockMat3 = idxRowMat.toHailBlockMatrix(10)
+    assert(blockMat2.rows === rows)
```

Comment:
These should be `blockMat3` not `blockMat2`.

Given comment above about testing more cases, I suggest you put this pattern into a `for` over each BlockSize